src/client/components/Auth.tsx

```diff
@@ -20,7 +20,7 @@ const Authorization = (allowedRoles: string[]) => (WrappedComponent: React.Compo
                 credentials: "same-origin",
             });
             let data = await result.json();
-            if (data.a) {
+            if (data.a >= 0 && data.a <= 4) {
```

Comment:
keep this DRY in case we add new roles later. `if (data.a >= 0 && data.a < userModel.length)`